src/parser/shaman/elemental/modules/talents/SurgeOfPower.js

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+
+import Analyzer from 'parser/core/Analyzer';
+import StatisticBox, { STATISTIC_ORDER } from 'interface/others/StatisticBox';
+import SpellIcon from 'common/SpellIcon';
+import SPELLS from 'common/SPELLS';
+import SpellLink from 'common/SpellLink';
+import { SELECTED_PLAYER } from 'parser/core/EventFilter';
+import Events from 'parser/core/Events';
+
+const SURGE_OF_POWER = {
+  INCREASE: 0.2,
+  DURATION: 15000,
+  WINDOW_DURATION: 500,
+  AFFECTED_CASTS: [
+    SPELLS.FLAME_SHOCK,
+    SPELLS.FROST_SHOCK,
+    SPELLS.LAVA_BURST,
+    SPELLS.LIGHTNING_BOLT,
+  ],
+  TALENTS: [
+    SPELLS.SURGE_OF_POWER_TALENT.id,
+  ],
+};
+
+class SurgeOfPower extends Analyzer {
+  sopBuffedAbilities = {};
+  sopActivationTimestamp = null;
+  sopConsumptionTimestamp = null;
+  // number of SoP empowered lava bursts
+  lvbCastCount = 0;
+  // % of SK lightning bolts empoered by SoP
+  sopSKCastsPercentage = 0.0;
+  // total SK + SoP lightning bolt casts
+  skSopCasts = 0;
+  // total SK lightning bolt casts
+  skCasts = 0;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTalent(SPELLS.SURGE_OF_POWER_TALENT.id);
+
+    for (const key in SURGE_OF_POWER.AFFECTED_CASTS) {
+      const spellid = SURGE_OF_POWER.AFFECTED_CASTS[key].id;
+      if((this.selectedCombatant.hasTalent(spellid)) || (!SURGE_OF_POWER.TALENTS.includes(spellid))) {
```

Comment:
same goes here
this does nothing

for mote it checks if one of the abilities is not talented into
so it returns false if the talent gets affected by mote but is not talented into

SoP does not affect any talents so you don't need the condition here